src/app/app.component.ts

```diff
@@ -1,7 +1,31 @@
-import { Component } from '@angular/core';
+import { UserService } from './user.service';
+import { AuthService } from './auth.service';
+import { Router } from '@angular/router';
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html'
 })
-export class AppComponent {}
+export class AppComponent implements OnDestroy {
+  userSubscription: Subscription = null;
```

Comment:
you can probably make this private since you are only using it in the scope of this file

Reply:
Done